src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotDirectlyAwaitATask.cs

```diff
@@ -78,22 +78,29 @@ public override void Initialize(AnalysisContext analysisContext)
 
         private static void AnalyzeOperation(OperationAnalysisContext context, ImmutableArray<INamedTypeSymbol> taskTypes)
         {
-            IAwaitOperation awaitExpression = context.Operation as IAwaitOperation;
+            var awaitExpression = (IAwaitOperation)context.Operation;
 
             // Get the type of the expression being awaited and check it's a task type.
-            ITypeSymbol typeOfAwaitedExpression = awaitExpression?.Operation?.Type;
+            ITypeSymbol? typeOfAwaitedExpression = awaitExpression.Operation.Type;
             if (typeOfAwaitedExpression != null && taskTypes.Contains(typeOfAwaitedExpression.OriginalDefinition))
             {
                 context.ReportDiagnostic(awaitExpression.Operation.Syntax.CreateDiagnostic(Rule));
             }
         }
 
-        private static ImmutableArray<INamedTypeSymbol> GetTaskTypes(Compilation compilation)
+        private static bool TryGetTaskTypes(Compilation compilation, [NotNullWhen(returnValue: true)] out ImmutableArray<INamedTypeSymbol> taskTypes)
         {
-            INamedTypeSymbol taskType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTask);
-            INamedTypeSymbol taskOfTType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksGenericTask);
+            INamedTypeSymbol? taskType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTask);
+            INamedTypeSymbol? taskOfTType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksGenericTask);
 
-            return ImmutableArray.Create(taskType, taskOfTType);
+            if (taskType == null || taskOfTType == null)
+            {
+                taskTypes = default;
```

Comment:
Might as well make the result a valid instance.

```suggestion
                taskTypes = ImmutableArray<INamedTypeSymbol>.Empty;
```